test/test_gradient.jl

```diff
@@ -77,7 +96,8 @@ NNLIB_ACTIVATIONS = union(NNLIB_ACTIVATIONS_S, NNLIB_ACTIVATIONS_F)
         end
 
         ## Error handling when applying non-dual tracers to "local" functions with control flow
-        @test_throws MissingPrimalError jacobian_sparsity(
+        # TypeError: non-boolean (SparseConnectivityTracer.GradientTracer{BitSet}) used in boolean context
+        @test_throws TypeError jacobian_sparsity(
```

Comment:
add this test to connectivity and hessian too

Reply:
Done.